internal/spec/spec.go

```diff
@@ -219,6 +219,7 @@ type RunnerSpec struct {
 	UseEphemeralStorage      bool
 	VirtualNetworkCIDR       string
 	UseAcceleratedNetworking bool
+	VnetSubnetID             string
```

Comment:
Could we add another flag here called `disable_isolated_networks` that defaults to `false`?

The idea is to allow for a flag that switches between the explicit `vnet_subnet_id` potentially added to a config and the isolated network model.

For example, the current behavior is to create isolated networks for each runner. Simply specifying the `vnet_subnet_id` in the config will mean that all pools will use it, even if they don't mean to.  A toggle would allow the operator to say:

```bash
garm-cli pool update pool_id --extra-specs='{"disable_isolated_networks": true}'
```

Which would automatically use the `vnet_subnet_id` in the config. Of course, if `disable_isolated_networks` is `true` and there is no `vnet_subnet_id` in the config, it would err out.

Reply:
I think it's okay to have this in the config as a valid default value. We can also enable it by default for all pools using the `disable_isolated_networks` feature flag.

From a UX perspective, by default it doesn't change current behavior by simply setting the option in the config, but does allow users to enable it if they want by default. It all depends on their use case.

Users may want it enabled for most of their pools without having to explicitly set it in extra specs for many pools, or they may want it disabled for most of their pools.

Adding an additional feature gate flag, allows us to create a config like:

```toml
vnet_subnet_id = "the subnet"
disable_isolated_networks = true
```

And every pool benefits from this by default, unless:

```bash
garm-cli pool update pool_id --extra-specs='{"disable_isolated_networks": false}'
```

OR, users may define the config like:

```toml
vnet_subnet_id = "the subnet"
disable_isolated_networks = false
```

in which case current behavior is kept, and users could enable this feature only on select pools:

```bash
garm-cli pool update pool_id --extra-specs='{"disable_isolated_networks": true}'
```

In both cases, they would benefit from a default `vnet_subnet_id`. They would not have to define the subnet (which is a really long string) for each pool.

But you folks will be using this quite a bit I imagine. If you believe it's better without the feature flag, feel free to mark this as "resolved".  I don't feel strongly about it.